schematic/store/synapse.py

```diff
@@ -1517,6 +1517,60 @@ async def store_async_annotation(self, annotation_dict: dict) -> Annotations:
         )
         return await annotation_class.store_async(synapse_client=self.syn)
 
+    def process_row_annotations(
+        self,
+        dmge: DataModelGraphExplorer,
+        metadata_syn: Dict[str, Any],
+        hide_blanks: bool,
+        csv_list_regex: str,
+        annos: Dict[str, Any],
+        annotation_keys: str,
+    ) -> Dict[str, Any]:
+        """processes metadata annotations
+
+        Args:
+            dmge (DataModelGraphExplorer): data model graph explorer
+            metadata_syn (dict): metadata used for Synapse storage
+            hideBlanks (bool): if true, does not upload annotation keys with blank values.
+            csv_list_regex (str): Regex to match with comma separated list
+            annos (Dict[str, Any]): dictionary of annotation returned from synapse
+            annotation_keys (str): display_label/class_label
+
+        Returns:
+            Dict[str, Any]: annotations as a dictionary
+        """
+        for anno_k, anno_v in metadata_syn.items():
+            # Remove keys with nan or empty string values or string that only contains white space from dict of annotations to be uploaded
+            # if present on current data annotation
+            if hide_blanks and (
+                (isinstance(anno_v, str) and anno_v.strip() == "")
+                or (isinstance(anno_v, float) and np.isnan(anno_v))
+            ):
+                annos.pop(anno_k) if anno_k in annos.keys() else annos
+                continue
+
+            # Otherwise save annotation as approrpriate
+            if isinstance(anno_v, float) and np.isnan(anno_v):
+                annos[anno_k] = ""
+                continue
+
+            # Handle strings that match the csv_list_regex and pass the validation rule
```

Comment:
Comments like this might be better within the docstring. This is a decent breakdown in the difference in purpose of the two and I found:



| Feature   | Comments                                                                                                                                  | Docstrings                                                                                                                                                                                                 |
|-----------|-------------------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| Purpose   | Explains code purpose, provides context, adds reminders, etc.                                                                             | Programmatically documents code; should explain what the code does, can list inputs, outputs, and returns.                                                                                                 |
| Syntax    | Single-line comments: <br> `#` followed by a blank space. <br> Multi-line comments: <br> text is enclosed by `"""`, or each line is preceded by a `#` followed by a space. | Both multi-line and one-liner docstrings are enclosed by `"""`.                                                                                                                                            |
| Location  | Any place that is not immediately 1) after another docstring or, 2) the initial string following the definition of a module, function, or class is a place where a comment may be inserted. | The initial literal string that follows the definition of a module, function, or class and the first string after some other docstring is considered a docstring.                                          |
| Scope     | Only present in Python source code, removed at compile-time.                                                                               | Can be accessed by various automated documentation generators in both source code and bytecode.                                                                                                            |

Reply:
Thanks for the suggestion Bryan! This makes sense. The if else logic in the original code was pretty crazy and hard to understand. I will provide a mermaid diagram for this function so that the logic is clearer.